metafix/src/test/java/org/metafacture/metafix/MetafixScriptTest.java

```diff
@@ -430,41 +429,41 @@ private void assertStrictness(final Metafix.Strictness strictness, final boolean
     }
 
     @Test
-    @MetafixToDo("See https://github.com/metafacture/metafacture-fix/pull/170")
```

Comment:
But it's not working, right? It dosen't skip any expression, because there is no exception.

Reply:
Yes, it does throw an exception (change `Metafix.Strictness.EXPRESSION` to `Metafix.Strictness.PROCESS` to see for yourself; that's what the other tests are confirming as well) and it skips the `upcase()` expression (values remain lowercase).

Reply:
Oh, so leaving the values lowercase, but including them, is the desired behavior?

Reply:
Right, well I do think it makes sense to leave the values untouched if something goes wrong, instead of removing them. I was just surprised because that was not the behavior of the original test (which passed in master, and I marked as TODO when merging). So +1 from me here.

Reply:
Agreed. This is also what Catmandu does (hbz/Catmandu@370ca93).

As for the cause: Previously, the field would be [removed](https://github.com/metafacture/metafacture-fix/blob/c25517aaa2958b0ae98420b0f83c9962f176c944/metafix/src/main/java/org/metafacture/metafix/FixPath.java#L149) before the value's type was [checked](https://github.com/metafacture/metafacture-fix/blob/c25517aaa2958b0ae98420b0f83c9962f176c944/metafix/src/main/java/org/metafacture/metafix/FixPath.java#L154); now the [check](https://github.com/metafacture/metafacture-fix/blob/0dfcfb86f725baa73989e6fac52018f9f7c4049e/metafix/src/main/java/org/metafacture/metafix/Record.java#L204) happens before the [removal](https://github.com/metafacture/metafacture-fix/blob/0dfcfb86f725baa73989e6fac52018f9f7c4049e/metafix/src/main/java/org/metafacture/metafix/Record.java#L220), so an exception due to an unexpected type prevents the value from being removed.